### PR TITLE
fix(list): selectedIndex 0 isn't working

### PR DESCRIPTION
**Pull request by TroyTae**
As you know, 0 is replaced with false in javascript.
When selectedIndex is 0, setSelectedIndex is not executed.
So I add check code "Number.isInteger".

### CODE REVIEW

**Comment by TroyTae** [comment]
I signed it!

File: packages/list/index.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default class List extends Component {
     this.foundation_.setSingleSelection(singleSelection);
     this.foundation_.setWrapFocus(wrapFocus);
     this.foundation_.setVerticalOrientation(this.props[ARIA_ORIENTATION] === VERTICAL);
-    if (selectedIndex) {
+    if (Number.isInteger(selectedIndex)) {
```
Comment (moog16) [inline packages/list/index.js]: `isInteger` is not supported by IE11, which is something we need to support. Try using the `typeof selectedNumber === 'number' && isNaN(selectedNumber)` check. I believe `typeof NaN` returns 'number', which is why we need the last part of that statement. 

Also please add a few unit tests to cover the case you mentioned in your first comment of this PR. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

Comment (TroyTae) [reply packages/list/index.js]: Oh I mistook it for typescript.
I will change it and write some test code!

**Comment by TroyTae** [comment]
@moog16 
Thank you for your verification.

**Comment by moog16** [comment]
Opened PR https://github.com/material-components/material-components-web-react/pull/406 for tests

**Comment by moog16** [comment]
tests passing...merging!

**Pull request merged**